fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/query/FDBCoveringIndexQueryTest.java

```diff
@@ -894,4 +905,110 @@ void coveringPrimaryKey() throws Exception {
         assertEquals(284123809, plan.planHash(PlanHashable.PlanHashKind.FOR_CONTINUATION));
         assertEquals(994460375, plan.planHash(PlanHashable.PlanHashKind.STRUCTURAL_WITHOUT_LITERALS));
     }
+
+    @Test
+    void coveringWideFunctionKey() throws Exception {
+        final Index nonFlattenedIndex = new Index("nonFlattenedIndex", concat(
+                field("num_value_2"),
+                function("non_flattened", concatenateFields("num_value_3_indexed", "str_value_indexed"))
```

Comment:
> Unless the way it is failing is that it was extracting the wrong positions for the primary key.

Correct, it fails with a type error on the primary key, trying to use `str_value_indexed` for `rec_no`, believing that the `non_flattened` field occupies only position `1`, rather than positions `1-2`:

```
Caused by: java.lang.IllegalArgumentException: Wrong object type used with protocol message reflection.
Field number: 1, field java type: LONG, value type: java.lang.String

	at com.google.protobuf.FieldSet$Builder.verifyType(FieldSet.java:1214)
	at com.google.protobuf.FieldSet$Builder.setField(FieldSet.java:1081)
	at com.google.protobuf.DynamicMessage$Builder.setField(DynamicMessage.java:545)
	at com.google.protobuf.DynamicMessage$Builder.setField(DynamicMessage.java:320)
	at com.apple.foundationdb.record.query.plan.IndexKeyValueToPartialRecord$FieldCopier.copy(IndexKeyValueToPartialRecord.java:208)
	at com.apple.foundationdb.record.query.plan.IndexKeyValueToPartialRecord.toRecordInternal(IndexKeyValueToPartialRecord.java:70)
	at com.apple.foundationdb.record.query.plan.IndexKeyValueToPartialRecord.toRecord(IndexKeyValueToPartialRecord.java:62)
	at com.apple.foundationdb.record.query.plan.plans.QueryPlanUtils.lambda$getCoveringIndexEntryToPartialRecordFunction$0(QueryPlanUtils.java:63)
	at com.apple.foundationdb.record.RecordCursorResult.map(RecordCursorResult.java:177)
	at com.apple.foundationdb.record.RecordCursor.lambda$map$6(RecordCursor.java:336)
```

The copier is `[num_value_2: KEY[0], rec_no: KEY[2]]` and the first index entry is `nonFlattenedIndex:(1, 0, "even", 10):()`.